client/model.go

```diff
@@ -851,20 +851,32 @@ type ServerInfo struct {
 	*responsev1.ServerInfoResponse
 }
 
+type AuditLogType uint8
+
+const (
+	AccessLogs AuditLogType = iota
+	DecisionLogs
+)
+
 // AuditLogOptions is used to filter audit logs.
 type AuditLogOptions struct {
+	Type      AuditLogType
 	Tail      uint32
 	StartTime time.Time
 	EndTime   time.Time
 	Lookup    string
 }
 
-type AccessLogEntry struct {
-	Log *auditv1.AccessLogEntry
-	Err error
+type AuditLogEntry struct {
+	accessLog   *auditv1.AccessLogEntry
+	decisionLog *auditv1.DecisionLogEntry
+	Err         error
+}
+
+func (e *AuditLogEntry) AccessLog() *auditv1.AccessLogEntry {
+	return e.accessLog
```

Comment:
I am not sure why `Err` is a public member but log entries are private and accessible only through methods. Maybe you were trying to do something like the following?

```suggestion
type AuditLogEntry struct {
	accessLog   *auditv1.AccessLogEntry
	decisionLog *auditv1.DecisionLogEntry
	err         error
}

func (e *AuditLogEntry) AccessLog() (*auditv1.AccessLogEntry, error) {
      if e.err != nil {
          return nil, err
      }
	return e.accessLog, nil
```

Reply:
Hmm, It is actually mean to be used like this:

```Go
if err := entry.Err; err != nil {
	// handle error
}
// do something with entry.AccessLog()
```

Since `error` is not the key value here I decided not to protect it as log entries. We can also protect it by making it private and add a method like we do for log entries.

Reply:
Is there a reason to protect log entries? The inconsistent mixing of members and methods  does not feel right to me. Given that users are supposed to check for errors on every log entry anyway, I think I prefer the suggestion I made above where the method returns the log entry or the error.

Reply:
Fair enough 👍